lib/rspec/core/metadata_filter.rb

```diff
@@ -13,24 +13,19 @@ def apply?(predicate, filters, metadata)
         end
 
         # @private
-        def filter_applies?(key, value, metadata)
+        def filter_applies?(key, filter_value, metadata)
           silence_metadata_example_group_deprecations do
-            return location_filter_applies?(value, metadata) if key == :locations
-            return id_filter_applies?(value, metadata)       if key == :ids
-            return filters_apply?(key, value, metadata)      if Hash === value
-
-            return false unless metadata.key?(key)
-            return true if TrueClass === value && !!metadata[key]
-            return filter_applies_to_any_value?(key, value, metadata) if Array === metadata[key] && !(Proc === value)
-
-            case value
-            when Regexp
-              metadata[key] =~ value
-            when Proc
-              proc_filter_applies?(key, value, metadata)
-            else
-              metadata[key].to_s == value.to_s
-            end
+            return location_filter_applies?(filter_value, metadata) if key == :locations
+            return id_filter_applies?(filter_value, metadata)       if key == :ids
+            return filters_apply?(key, filter_value, metadata)      if Hash === filter_value
+
+            meta_value = metadata.fetch(key) { return false }
+
+            return true if TrueClass === filter_value && !!meta_value
```

Comment:
> to me it looks like a no-op unless the return value of meta_value can be a non-boolean value.

It can be a non-boolean value, but thats not an issue if we don't leak it.